content/doc/book/system-administration/obtaining-a-thread-dump.adoc

```diff
@@ -0,0 +1,67 @@
+---
+layout: section
+wip: true
+---
+ifdef::backend-html5[]
+:notitle:
+:description:
+:author:
+:email: jenkinsci-users@googlegroups.com
+ifdef::env-github[:imagesdir: ../resources]
+ifndef::env-github[:imagesdir: ../../resources]
+:toc: left
+endif::[]
+
+= Obtaining a thread dump
+
+During the troubleshooting of Jenkins, others may request that you obtain
+thread dumps of relevant Java VMs. Thread dumps concisely capture what
+every thread in a VM is doing at a given point in time.
+They are useful to diagnose hang problems, deadlocks, and performance issues.
+This page explains how you can obtain a thread dump.
+
+[[Obtainingathreaddump-FromJenkinsWebUI]]
+== From Jenkins Web UI
+
+This is the simplest way of obtaining thread dumps.
+
+If Jenkins or its build agents are operating normally, you can obtain a
+thread dump remotely by going to
+`+http://your.jenkins.server/threadDump+`. For an agent named 'xyz', go
+to `+http://your.jenkins.server/computer/xyz/systemInfo+`. You need to
+have the administrator permission on the system.
+
+[[Obtainingathreaddump-Byusingjstack]]
+== By using `+jstack+`
+
+If Jenkins is not responding to web UI, try
+https://docs.oracle.com/javase/8/docs/technotes/guides/troubleshoot/tooldescr016.html[jstack]
+to obtain the thread dump. For one reason or another that none of us
+quite understand well, one might have to add -F to get the dump. If that
+was the case, please mention that in the report as well.
+
+Make sure to run jstack as the same user that's running Jenkins itself
+instead of using the root user.
+
+[[Obtainingathreaddump-Bysendingsignal]]
+== By sending signal
+
+If the above two approaches do not work, you can still have the JVM
+print the thread dump to its stdout by sending it a signal.
+
+If you have a terminal or command prompt that's running the JVM, you can
+hit Ctrl+ + (Unix) or Ctrl+Break (Windows) to do this. If the JVM is running in
+background, you do this by `+kill -3 PID+` (Unix) or use
+https://docs.oracle.com/javacomponents/jmc-5-5/jmc-user-guide/toc.htm/[a tool like
+this] to send a signal (Windows).
```

Comment:
Generally prefer to not split links across a line because it confuses the syntax highlighting in my editor.
```suggestion
https://docs.oracle.com/javacomponents/jmc-5-5/jmc-user-guide/toc.htm/[a tool like this] to send a signal (Windows).
```